src/App.tsx

```diff
@@ -55,6 +55,26 @@ class App extends React.Component<StyledComponentProps, {}> {
             <Route path="/catalog/:definitionId">
               <Catalog catalogDetailMode="introduction" />
             </Route>
+            <Route
+              path="/_catalog/:definitionId/keymap"
+              render={(props) => {
+                const path = `/catalog/${props.match.params.definitionId}/keymap${props.location.search}`;
+                props.history.push(path);
+                return null;
+              }}
+            />
+            <Redirect
+              from="/_catalog/:definitionId"
+              to="/catalog/:definitionId"
+            />
+            <Route
+              path="/_catalog"
+              render={(props) => {
+                const path = `/catalog${props.location.search}`;
+                props.history.push(path);
+                return null;
+              }}
+            />
```

Comment:
By these routing definitions, the new path `/_catalog` is added. These `/_catgalog` paths are accessed from the index.html content sent from the Firebase Functions. For instance, the index.html content has the script to redirect to the `/_catalog` path (`location.href = '/_catalog/...';`). Here, I used the `render` attribute, because the `<Redirect>` tag doesn't pass any query parameters. Therefore, to pass through all query parameters, I wrote the logic using `props.history.push` with the `location.search` value.